test/unittests/test_cli.py

```diff
@@ -0,0 +1,157 @@
+                "https://iiif.test.dasch.swiss",
+            ],
+            "http://app.staging.dasch.swiss/": [
+                "https://api.staging.dasch.swiss",
+                "https://iiif.staging.dasch.swiss",
+            ],
+            "https://demo.dasch.swiss/": [
+                "https://api.demo.dasch.swiss",
+                "https://iiif.demo.dasch.swiss",
+            ],
+            "http://api.dev.dasch.swiss/": [
+                "https://api.dev.dasch.swiss",
+                "https://iiif.dev.dasch.swiss",
+            ],
+            "dev-02.dasch.swiss": [
+                "https://api.dev-02.dasch.swiss",
+                "https://iiif.dev-02.dasch.swiss",
+            ],
+            "082a-test-server.dasch.swiss": [
+                "https://api.082a-test-server.dasch.swiss",
+                "https://iiif.082a-test-server.dasch.swiss",
+            ],
+            "admin.08F4-test-server.dasch.swiss": [
+                "https://api.08F4-test-server.dasch.swiss",
+                "https://iiif.08F4-test-server.dasch.swiss",
+            ],
+            "app.08F4-test-server.dasch.swiss": [
+                "https://api.08F4-test-server.dasch.swiss",
+                "https://iiif.08F4-test-server.dasch.swiss",
+            ],
+            "iiif.E5bC-test-server.dasch.swiss": [
+                "https://api.E5bC-test-server.dasch.swiss",
+                "https://iiif.E5bC-test-server.dasch.swiss",
+            ],
+            "not-yet-0826-test-server.dasch.swiss": [
+                "https://api.not-yet-0826-test-server.dasch.swiss",
+                "https://iiif.not-yet-0826-test-server.dasch.swiss",
+            ],
+            "https://admin.ls-prod-server.dasch.swiss": [
+                "https://api.ls-prod-server.dasch.swiss",
+                "https://iiif.ls-prod-server.dasch.swiss",
+            ],
+            "https://ls-test-server.dasch.swiss": [
+                "https://api.ls-test-server.dasch.swiss",
+                "https://iiif.ls-test-server.dasch.swiss",
+            ],
+        }
+
+    def test_derive_sipi_url(self) -> None:
+        """
+        Test the method that derives the SIPI URL from the DSP API URL
+        """
+        for api_url_orig, expected in self.positive_testcases.items():
+            api_url_canonical, sipi_url = expected
+            args_orig = argparse.Namespace(
+                action="xmlupload",
+                server=api_url_orig,
+                user="some.user@dasch.swiss",
+                password="password",
+                xmlfile="data.xml",
+            )
+            args_new = _derive_sipi_url(
+                parsed_arguments=args_orig,
+                default_dsp_api_url=self.default_dsp_api_url,
+                default_sipi_url=self.default_sipi_url,
+            )
+            self.assertEqual(args_new.server, api_url_canonical)
+            self.assertEqual(args_new.sipi_url, sipi_url)  # pylint: disable=no-member
```

Comment:
see my comment above on the information flow: if `_derive_sipi_url()` is changed to a `str -> tuple[str, str]` function, then this test can be simplified a lot:

```python
        api, sipi = _derive_sipi_url("localhost:3333") # or whatever you want to test
        self.assertEqual(api, "http://0.0.0.0:3333")
        self.assertEqual(sipi, "http://0.0.0.0:1024")
```

and as such it's also not too much to have separate test methods:

```python
    def test_derive_sipi_localhost(self) -> None:
        api, sipi = _derive_sipi_url("localhost:3333")
        self.assertEqual(api, "http://0.0.0.0:3333")
        self.assertEqual(sipi, "http://0.0.0.0:1024")

    def test_derive_sipi_localhost_zeros(self) -> None:
        api, sipi = _derive_sipi_url("0.0.0.0:3333")
        self.assertEqual(api, "http://0.0.0.0:3333")
        self.assertEqual(sipi, "http://0.0.0.0:1024")
```

etc.